beautyjoy/BJC-Teacher-Tracker
# Fix email template reseed problem

### [Pivotal Tracker Link][tracker]


[tracker]: https://www.pivotaltracker.com/story/show/187113121

## What this PR does:


This pull request ___fixes___ the issue of duplication and deletion of Email Templates upon reseeding the database with updated seed data.

### Include screenshots, videos, etc.
#### Previous Problem - Duplication of Email Templates after Reseeding Updated Email Templates
![Previous Problem](https://github.com/beautyjoy/BJC-Teacher-Tracker/assets/92573736/3091bd23-73fa-407a-a5d2-a98b2a8f9b2d)
#### After Fix
1. No Duplication of Email Templates after Reseeding Updated Email Templates.
2. It also considers the case to completely delete the email template and reseed it again, ensuring the database reflects the most current seed data without orphaned records.

#### Who authored this PR?

1. Perry (Jingchao) Zhong
2. Michael Tao

### How should this PR be tested?

- Ensure that the test suite covers scenarios for both updating existing email templates and removing obsolete templates from the database upon reseeding.
- Specifically, the tests should verify that after reseeding:
    1. Updated templates reflect the new changes without creating duplicate entries.
    2. Templates not included in the new seed data are properly removed from the database.

* **What do the specs/features test?**
  The specs test the reseeding process's ability to update and clean up email templates in the database accurately.
* **Are there edge cases to watch out for?**
  Not yet.

#### Are there any complications to deploying this?
Possible to have. Potential complications include the deletion of email template records from the database, aiming to prevent duplication and remove outdated templates. This can lead to data loss if users rely on templates not included in the seed data but present in the database, unaware of these changes.

### Checklist:

- [x] I have renamed the branch to match PivotTracker's suggested one (necessary for BlueJay) (e.g., `perry/187113121-fix-email-template-reseed-issue`). Any branch name will do as long as the story ID is there. You can use `git checkout -b [new-branch-name]`
